packages/core/src/hooks/useSigner.tsx

```diff
@@ -0,0 +1,15 @@
+import { useEthers } from './useEthers'
+import { useState, useEffect } from 'react'
+import { JsonRpcProvider, JsonRpcSigner } from '@ethersproject/providers'
+
+export function useSigner(): JsonRpcSigner | undefined {
+  const { library, account } = useEthers()
+  const [signer, setSigner] = useState<JsonRpcSigner | undefined>()
+
+  useEffect(() => {
+    if (account) setSigner((library as JsonRpcProvider).getSigner())
```

Comment:
It might be safer to do:

```
if (library !== undefined && 'getSigner' in library)
```

Because if for some reason `account` would update faster than `library` you would have a problem.
And this way TS is smart enough and you don't have to use `as`